### DISCUSSION TITLE
Upgrading to rdf4j 4.0.1 NoClassDefFound org/eclipse/rdf4j/OpenRDFUtil

### DISCUSSION THREAD

**Discussion by lewismc** [eclipse-rdf4j/rdf4j#3916] [Q&A] [answered]
In attempting to upgrade [Apache Any23 to use rdf4j 4.0.1](https://github.com/apache/any23/pull/288) I experienced a NoClassDefFound org/eclipse/rdf4j/OpenRDFUtil as a result of the owlapi libraries not being able to find the absent [OpenRDFUtil](https://github.com/eclipse/rdf4j/blob/3.7.7/core/model/src/main/java/org/eclipse/rdf4j/OpenRDFUtil.java) which was previously available in rdf4j 3.7.X.
```
[ERROR] Errors: 
[ERROR]   FunctionalSyntaxExtractorTest.testExampleFunctionalSyntax:49->AbstractExtractorTestCase.assertExtract:230->AbstractExtractorTestCase.assertExtract:212->AbstractExtractorTestCase.extract:199 » NoClassDefFound org/eclipse/rdf4j/OpenRDFUtil
[ERROR]   ManchesterSyntaxExtractorTest.testExampleManchesterSyntax:67 » NoClassDefFound org/eclipse/rdf4j/OpenRDFUtil
```
Having looked at the rdf4j javadoc I can see that it has OpenRDFUtil indeed been removed... so I may just postpone the upgrade to rdf4j. I just wanted to check here first.
Thanks for any information.

**Comment by hmottestad** [comment]
Yes it has been removed. Here is the relevant PR: https://github.com/eclipse/rdf4j/pull/3225

**Comment by abrokenjester** [comment, marked as answer]
Like @hmottestad pointed out it has indeed been removed. It looks like I did not include this in the upgrade notes, apologies for that. 

The only method this class offered was a `verifyContextNotNull` method:

```java
public static void verifyContextNotNull(Resource... contexts) 
```

...which threw an exception when the supplied input param is null, and was used solely to check validity of the `context` vararg in RDF4J's public methods.

In RDF4J itself its use has just been replaced by:

```java
Objects.requireNonNull(contexts,
				"contexts argument may not be null; either the value should be cast to Resource or an empty array should be supplied");
```

It shouldn't be hard to do the same replacement in the Any23 code.

**Comment by lewismc** [comment]
Thank you both @hmottestad @jeenbroekstra I'll go and submit a PR in the [owlapi project](https://github.com/owlcs/owlapi/search?q=OpenRDFUtil).

**Reply by lewismc** [reply]
https://github.com/owlcs/owlapi/pull/1065
